app/lib/screens/mess/manager/complaints_mess_manager.dart

```diff
@@ -1,30 +1,77 @@
+import 'dart:convert';
+
 import 'package:flutter/material.dart';
+import 'package:flutter_flavor/flutter_flavor.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:ira/screens/mess/manager/feedback_model.dart';
+import 'package:http/http.dart' as http;
 
 class ComplaintMessManager extends StatefulWidget {
-  const ComplaintMessManager({Key? key}) : super(key: key);
+  ComplaintMessManager({Key? key}) : super(key: key);
+  final secureStorage = const FlutterSecureStorage();
+  final String baseUrl = FlavorConfig.instance.variables['baseUrl'];
 
   @override
   State<ComplaintMessManager> createState() => _ComplaintMessManagerState();
 }
 
 class _ComplaintMessManagerState extends State<ComplaintMessManager> {
-  bool _actionTaken = false;
+  @override
   final List<String> _filter = ["Filter", "Date", "Mess", "Action"];
   String _filterValue = "Filter";
 
-  @override
+  Future<List<FeedbackModel>> _getMessFeedbackItems() async {
+    final String? token = await widget.secureStorage.read(key: 'staffToken');
+    final requestUrl = Uri.parse(widget.baseUrl + '/mess/feedback');
+    final response = await http.get(
+      requestUrl,
+      headers: <String, String>{
+        "Content-Type": "application/x-www-form-urlencoded",
+        'Authorization': token != null ? 'Token ' + token : '',
+      },
+    );
+
+    if (response.statusCode == 200) {
+      final data = jsonDecode(response.body);
+      return data
+          .map<FeedbackModel>((json) => FeedbackModel.fromJson(json))
+          .toList();
+    } else {
+      throw Exception('Failed to load post');
```

Comment:
Instead show an error to user that there is a problem with network connection. This can be done later.